app/controllers/admin/products_controller.rb

```diff
@@ -20,40 +22,17 @@ def product_params_without_nested
     product_params.except(:certifications_attributes, :nutritions_attributes, :packagings_attributes)
   end
 
-  def product_success(action)
-    action = addE(action)
-    flash[:message] = "#{action.capitalize}d Product"
-    flash[:type] = 'alert alert-success'
-    redirect_to admin_products_path
-  end
-
-  def product_fail(action, product)
-    flash[:message] = product.errors.full_messages
-    flash[:type] = 'alert alert-danger'
-    if action == "add"
-      flash[:product_params] = product_params
-      redirect_to new_admin_product_path
-    elsif action == "update"
-      redirect_to edit_admin_product_path
-    end
-  end
-
   def new
     # Make new product so form knows to make submit button say "Create Product"
     # Pass in params from form if redirected from #create
-    @product = Product.new(flash[:product_params])
+    @product = Product.new(flash[:prev_params])
```

Comment:
what is `prev_params` in this context?

Reply:
`prev_params` in this case are the params passed in to `create` before it failed and got redirected to `new`. The purpose of this is to make sure the information in the form does not get lost if a validation fails, for example.